core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp

```diff
@@ -168,7 +168,8 @@ struct ParallelReduceSpecialize<FunctorType, Kokkos::RangePolicy<PolicyArgs...>,
     const auto begin = p.begin();
     const auto end   = p.end();
 
-    ValueType result = ValueType();
+    ValueType result;
+    OpenMPTargetReducerWrapper<ReducerType>::init(result);
```

Comment:
The same initialization happens L179 after the `omp declare reduction` clause

Reply:
> The same initialization happens L179 after the `omp declare reduction` clause

Fixed it.